server/aws-lsp-json/src/language-server/jsonServer.ts

```diff
@@ -133,22 +133,9 @@ export const JsonServerFactory =
         }
     }
 
-async function getSchema(url: string) {
-    const response = await fetch(url)
-    const schema = await (await response.blob()).text()
-
-    return schema
-}
-
 export const CreateJsonLanguageServer = (
     defaultSchemaUri: string,
-    allowComments: boolean = true,
-    uriResolver: (url: string) => Promise<string> = getSchema
-) =>
-    JsonServerFactory(
-        create({
-            defaultSchemaUri: defaultSchemaUri,
-            allowComments: allowComments,
-            uriResolver: uriResolver,
-        })
-    )
+    customServiceClass?: JsonLanguageService,
```

Comment:
Since a) this approach breaks backwards incompatibility and b) a customServiceClass means that the other props would no longer be applicable—even though for example `defaultSchemUri` would still be mandatory—, I think a nice alternative here is to export a new function instead:

```ts
export function createCustomJsonLanguageServer(customService: JsonLanguageService) {
    return JsonServerFactory(new MutuallyExclusiveLanguageService([customService]));
}
```

Reply:
Thanks for the callout - Updated to use the alternative approach with a new method for when a custom language service is needed.